rllib/examples/custom_metrics_and_callbacks.py

```diff
@@ -78,12 +78,16 @@ def on_episode_end(
         env_index: int,
         **kwargs
     ):
-        # Make sure this episode is really done.
-        assert episode.batch_builder.policy_collectors["default_policy"].batches[-1][
-            "dones"
-        ][-1], (
-            "ERROR: `on_episode_end()` should only be called " "after episode is done!"
-        )
+        # Check if there are multiple episodes in a batch, i.e.
```

Comment:
thanks a ton for noticing this.
any suggestion on how can we assert this for "complete_episode" mode as well?

Reply:
# Trainer class
I think in case of `complete_episodes` we do not have to care about, if episodes are really `done` because they are always done (__complete__). But to have a callback that can run with both `batch_mode`s we need a prior safeguard for this. In case of `truncate_episodes` we have to be careful if the episode is indeed done. 

# Custom usage
This is different, if a user implements the callback in his own code and calls this callback before an episode is done. I don't know if (1) this case is rare and would happen only between a rollout and a batch completion or (2) if it can happen literally at any point of sampling. If we want to ensure that the below assertion does not error out, we might have a look at the `batches` attribute of the `PolicyCollector` as this gets emptied when the `MultiAgentBatch` is created by the `simple_list_collector`. I have also not yet figured out, if this attribute (`batches`) is only emptied, when the episode is `done`.

Reply:
Actually, even in "complete_episodes" mode, there could be more than one episodes in the final train batch (but not here in this callback).
* We should probably rename `multiple_episodes_in_batch` into `collect_complete_episodes` (bool).
* I think we should rather use `worker.policy_config["batch_mode"]` here in this if block.

Reply:
Hey @simonsays1980 , thanks for this fix! Let me know, if you want to do a follow-up PR with the 2 above suggested changes. I think this would help clarify these batch generating rules for the different settings even more.

Reply:
Hi @sven1977, yes I can do a follow-up PR. Should I do this within the same branch or better create a new one?